distribution/pull_v2.go

```diff
@@ -367,13 +370,35 @@ func (p *v2Puller) pullV2Tag(ctx context.Context, ref reference.Named, platform
 			"remote": ref,
 		}))
 
-	manifest, err := p.manifestStore.Get(ctx, specs.Descriptor{
+	desc := specs.Descriptor{
 		MediaType: mt,
 		Digest:    dgst,
 		Size:      size,
-	})
+	}
+	manifest, err := p.manifestStore.Get(ctx, desc)
 	if err != nil {
-		return false, err
+		if isTagged && isNotFound(errors.Cause(err)) {
+			logrus.WithField("ref", ref).WithError(err).Debug("Falling back to pull manifest by tag")
+
+			progress.Message(p.config.ProgressOutput, tagOrDigest, "DEPRECATED: Failed to pull manifest by the resolved digest")
+			progress.Message(p.config.ProgressOutput, tagOrDigest, "DEPRECATED: This registry does not appear to conform to the distribution specificiation")
```

Comment:
typo in specification. Also wonder if we should point out what's missing in the spec. https://github.com/docker/distribution/blob/master/docs/spec/api.md#pulling-an-image